### ISSUE TITLE
How to avoid unbalanced splitting using scaffold splitter?

### ISSUE THREAD

**Issue by hjzhang1018** [deepchem/deepchem#2187]
Description:
Binary classification dataset
RF model and GCNN model

I used `scaffold splitter` to split my dataset into train_dataset, valid_dataset and test_dataset according to recommendations. However this splitter always causes unbalanced dataset, which means "there is only one class present in y_true" in valid/test dataset. Thus evaluation metrics like auc_roc_score/recall_score/precision_score are not able to work.

However when I use `random splitter`, my model works normally and computes evaluations normally. But when I tried `scaffold splitter`, aforementioned issue still happen. I guess it's because this splitter automatically collects compounds with similar structures into one dataset and these compounds have the same label (all 0s or 1s).

I wonder if there's way to avoid such splitting situation? Any help is appreciated!

**Comment by rbharath** [comment]
This is a good question! One possibility is you could try to increase the size of valid/test (say 60/20/20 instead of 80/10/10 train/valid/test split). This could allow you to get more samples in valid and possibly avoid the errors that you're seeing

**Comment by hjzhang1018** [comment]
Thank you!@rbharath! Yes this is a good way to avoid such errors, I tried different splitting fractions and it worked :)
